src/accessibility/browser.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  getRole,
+  computeAccessibleName,
+  computeAccessibleDescription,
+} from 'dom-accessibility-api';
+import type { AccessibilityTreeOptions } from '.';
+
+const indent = (text: string, indenter = '  ') =>
+  indenter + text.split('\n').join(`\n${indenter}`);
+
+const enum AccessibilityState {
+  /** This element is accessible and its descendents could be accessible */
+  Accessible,
+  /** This element is innaccessible, but its descendents could be accessible */
+  InaccessibleSelf,
+  /** Both this element and its descendents are inaccessible */
+  InaccessibleSelfAndDescendents,
+}
```

Comment:
I'm a little confused by these names. I'm used to thinking about an experience being somewhere on the accessibility spectrum, so it's not obvious to me what it would mean for an element to be fully accessible or inaccessible.

Does accessible in this context mean that the element is in the accessibility tree? I wonder if we could rename these options to make that clearer? (This may just be me. Curious what others think)

Reply:
> Does accessible in this context mean that the element is in the accessibility tree

Yes. I mean accessible as "can be accessed". Do you have a suggestion of what to name it? Looking through [the spec](https://www.w3.org/TR/wai-aria-1.2/#tree_exclusion) for language to use...

- Excluded/Included
- Hidden

Reply:
I'm not sure... `Excluded`/`Included` seems good. This is also a total nit so if you think what you have now is clearer that's fine by me.